Config update for msi

JanPrusinowski opened the pull request.

philipanda committed:
scripts/lib/robot.sh: Make execute-robot pass additional args

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
scripts/lib/robot.sh: Make execute-robot pass additional args

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

JanPrusinowski committed:
msi-z690-common.robot: enabled execution of CCC tests

Signed-off-by: Jan Prusinowski <jan.prusinowski@3mdeb.com>

JanPrusinowski committed:
menus.robot: Added turning on serial redirection in pikvm machines after
reset to defaults tianocore

Signed-off-by: Jan Prusinowski <jan.prusinowski@3mdeb.com>

JanPrusinowski committed:
apu-configuration-menu.robot: changed teardown to leave machine in
powered on state

Signed-off-by: Jan Prusinowski <jan.prusinowski@3mdeb.com>

JanPrusinowski committed:
menus.robot: Added turning on serial redirection after reset to defaults
on machines using pikvm

Signed-off-by: Jan Prusinowski <jan.prusinowski@3mdeb.com>

JanPrusinowski committed:
cpu-cores-count.robot: Added Power On options to make tests run more
stable

Signed-off-by: Jan Prusinowski <jan.prusinowski@3mdeb.com>

JanPrusinowski committed:
msi-pro-z690-a-wifi-ddr4.robot: added cpu and changed audio options

Signed-off-by: Jan Prusinowski <jan.prusinowski@3mdeb.com>

JanPrusinowski committed:
cpu-cores-count.robot: Changed power on to power cycle on as machine
wouldnt boot when using pikvm

Signed-off-by: Jan Prusinowski <jan.prusinowski@3mdeb.com>

JanPrusinowski committed:
msi-z690-common.robot: Added Power Cycle On to fix freeze on Power On on
MSI

Signed-off-by: Jan Prusinowski <jan.prusinowski@3mdeb.com>

JanPrusinowski committed:
msi-pro-z690-a-wifi-ddr4.robot: fixes DEF_CORES_PER_SOCKET to correct
value

Signed-off-by: Jan Prusinowski <jan.prusinowski@3mdeb.com>

JanPrusinowski committed:
verified-boot, uefi-setup-menu: added support for pikvm enabled machines

Signed-off-by: Jan Prusinowski <jan.prusinowski@3mdeb.com>

JanPrusinowski committed:
measured-boot: added support for pikvm machines

Signed-off-by: Jan Prusinowski <jan.prusinowski@3mdeb.com>

JanPrusinowski committed:
ecure-boot-lib.robot: cut line for bluetooth in check secure boot in
linux

Signed-off-by: Jan Prusinowski <jan.prusinowski@3mdeb.com>

JanPrusinowski committed:
tcg-opal-disk-password.robot: fixed passing arguments

Signed-off-by: Jan Prusinowski <jan.prusinowski@3mdeb.com>

JanPrusinowski committed:
keywords.robot: Removed not working option for pikvm causing fail in
usb-stack tests

Signed-off-by: Jan Prusinowski <jan.prusinowski@3mdeb.com>

krystian-hebel commented on lib/secure-boot-lib.robot:
```diff
@@ -148,7 +148,7 @@ Check Secure Boot In Linux
     # or just:
     # Secure boot disabled
-    ${out}=    Execute Command In Terminal    dmesg | grep "Secure boot"
+    ${out}=    Execute Command In Terminal    dmesg | grep "Secure boot" | grep -v "Bluetooth"
```
Nice catch, but please add a comment why this is needed, otherwise someone may try to remove it in the future.

macpijan replied:
@JanPrusinowski Likely we need this to be ported to: https://github.com/Dasharo/open-source-firmware-validation/commits/regression-msi/

BeataZdunczyk replied:
@JanPrusinowski please link where changes were added.

JanPrusinowski replied:
https://github.com/Dasharo/open-source-firmware-validation/commit/133b843580206f791fd72831bbb26e02ecfbaf75

krystian-hebel commented on platform-configs/include/msi-z690-common.robot:
```diff
@@ -128,3 +131,4 @@ Power On
     # read the old output
     Telnet.Read
     Rte Power On
+    Power Cycle On
```
Is this the correct order? I'd expect it to be before `Rte Power On`. You first have to plug the power cord, and then press the button.

JanPrusinowski replied:
This was fixed in https://github.com/Dasharo/open-source-firmware-validation/commits/regression-msi/

krystian-hebel commented on platform-configs/msi-pro-z690-a-wifi-ddr4.robot:
```diff
@@ -17,3 +17,15 @@ ${WIRELESS_CARD_BLUETOOTH_SUPPORT}=     ${TRUE}
+
+${DEVICE_AUDIO2}=                       Raptorlake HDMI
+
+${DEF_THREADS_PER_CORE}=                2
+${DEF_THREADS_TOTAL}=                   28
+${DEF_ONLINE_CPU}=                      0-27
+${DEF_SOCKETS}=                         1
+
+${DEF_CORES_PER_SOCKET}=                20
+
+${CPU_P_CORES_MAX}=                     8
+${CPU_E_CORES_MAX}=                     12
```
We have two such boards in the lab, but these are valid for only one of them. For now, please add a comment mentioning this fact and CPU model for which those values are valid, we will have to somehow handle it in the future. The other platform isn't fully remotely available anyway.

BeataZdunczyk replied:
@JanPrusinowski broken link ^

JanPrusinowski replied:
Works for me... But I edited it so that it is wraped.

krystian-hebel commented on lib/options/uefi-setup-menu.robot:
```diff
@@ -129,3 +129,18 @@ Measure Reboot Time
     ...    Calculate Boot Time Statistics    ${durations}
     RETURN    ${min}    ${max}    ${average}    ${stddev}
+
+Make Sure That Flash Locks Are Disabled
```
In fact, `Measure * Time` keywords also don't fit here :thinking:

JanPrusinowski replied:
It is copied from `lib/options/dcu.robot`. As I understand this file is included when options are changed by DCU. When not the `uefi-setup-menu.robot` is included. This command should be mirrored in both files... This kwd might not fit here but it is a DCU issue... As you noticed - it is not the only kwd which shouldnt be there. DCU might need a refactor to make order with those KWDs.

BeataZdunczyk replied:
If we don't want to resolve this issue here, maybe we could create an issue for it @JanPrusinowski ?

JanPrusinowski replied:
Issue: https://github.com/Dasharo/open-source-firmware-validation/issues/547

krystian-hebel commented on dasharo-compatibility/cpu-cores-count.robot:
```diff
@@ -58,6 +58,7 @@ CCC003.001 Check core count (HT Enabled, P: All, E: 0) (Ubuntu)
     Reset To Defaults Tianocore
     Save Changes And Reset
     Set UEFI Option    ActiveECores    0
+    Power Cycle On
```
(applies to all changes in this file)

Why is this needed? `Set UEFI Option` ends with `Save Changes And Reset`, is it broken?

macpijan replied:
Info: resolved by: https://github.com/Dasharo/open-source-firmware-validation/commit/836169f6940ec10d65c81c4c3a302f8c06905068

JanPrusinowski committed:
msi-pro-z690-a-wifi-ddr4.robot: added info about WiFi card in Ubuntu

Signed-off-by: Jan Prusinowski <jan.prusinowski@3mdeb.com>

JanPrusinowski committed:
msi0pro-z690-a-wifi-ddr4.robot: changed FW ver

Signed-off-by: Jan Prusinowski <jan.prusinowski@3mdeb.com>

JanPrusinowski committed:
msi-pro-z690-a-wifi-ddr4.robot: edited release date

Signed-off-by: Jan Prusinowski <jan.prusinowski@3mdeb.com>

JanPrusinowski committed:
msi-z690-common.robot: Added support for Reset To Default Tests

Signed-off-by: Jan Prusinowski <jan.prusinowski@3mdeb.com>

JanPrusinowski committed:
menus.robot: Reworked Reset To Defaults Tianocore for PiKVM. To restore
Serial Redirection

Signed-off-by: Jan Prusinowski <jan.prusinowski@3mdeb.com>

JanPrusinowski committed:
esp-scanning-lib.robot: Fixed Prepare EFI Partition With System Files
not working on PiKVM

Signed-off-by: Jan Prusinowski <jan.prusinowski@3mdeb.com>

JanPrusinowski committed:
esp-scanning: Restored previousely removed kwds to fix tests
esp-scanning-lib.robot: Fixed Clear Out EFI Partition on PiKVM

Signed-off-by: Jan Prusinowski <jan.prusinowski@3mdeb.com>

macpijan commented on keywords.robot:
```diff
@@ -1729,14 +1733,10 @@ Check That USB Devices Are Detected
     ...    assumes that a bootable image is hosted on the PiKVM already.
     [Arguments]    ${boot_menu}
 
-    IF    '${DUT_CONNECTION_METHOD}' == 'pikvm'
```
@krystian-hebel For the devices with PiKVM we simply look for the PiKVM device in the boot menu to evaluate whether the boot device is detected. For others, we have to use other ways, which may or may not work presently.

macpijan replied:
@JanPrusinowski Likely we need this to be ported to: https://github.com/Dasharo/open-source-firmware-validation/commits/regression-msi/

macpijan commented on lib/bios/menus.robot:
```diff
@@ -412,6 +413,48 @@ Reset To Defaults Tianocore
     Read From Terminal Until    ignore.
     Write Bare Into Terminal    y
 
+    # For platforms using PiKVM there is a need to turn on Serial Redirection otherwise most of tests might break.
```
I do not get it. If serial is not enabled, how this code can work, as it reads from serial? This seems pointless, we simply have to modify the test binary via DCU prior running tests, which is already documented.

Unless I'm missing something here, then please correct me.

macpijan replied:
Ok, I think I do get it on the second review. We want to enable serial after we reset settings to default, to ensure the serial remains enabled, even if it has just been disabled by resetting.

krystian-hebel replied:
This is part of `Reset To Defaults`, and default is disabled. DCU doesn't change the default, only the current setting.

macpijan replied:
If this is the case, it is not necessarily tied to PiKVM directly in any way.

This is simply needed for devices where serial is disabled by default. Which may or may not be related to the fact that they do have PiKVM connected as well.

What is more, it should not hurt to make sure this options is selected even on devices where serial is enabled after resetting to defaults.

JanPrusinowski replied:
Fortunately resetting options to default does not turn off serial redirection until DUT is reset. So the option can be turned on before so.
My understanding was that atm only PiKVM enabled platforms use serial redirection. However it is a good idea to differentiate where this option should be enabled by some other way.

BeataZdunczyk replied:
@JanPrusinowski please link the issue here

JanPrusinowski committed:
esp-scanning-lib.robot: No removing Windows files from efi partition

Signed-off-by: Jan Prusinowski <jan.prusinowski@3mdeb.com>

JanPrusinowski committed:
esp-scanning.robot: Fixed false negative in ESP005.001

Signed-off-by: Jan Prusinowski <jan.prusinowski@3mdeb.com>

JanPrusinowski committed:
uefi-setup-menu.robot: fixed get uefi option by removing unnecesary
argument
rebar.robot: changed check for the test to fit changes in get uefi
option

Signed-off-by: Jan Prusinowski <jan.prusinowski@3mdeb.com>

macpijan commented on lib/bios/menus.robot:
```diff
@@ -412,6 +413,48 @@ Reset To Defaults Tianocore
+        END
+    END
+
+Return Highlight To Default Position In Setup Menu
```
I checked and it does not work for me, it sets highlight to the `Reset` instead

macpijan replied:
The problem is, that after 1st ESC what you get over telnet when reading until `save screenshot` is the menu, but with no help text visible:

```
QEMU x86 q35/ich9
 QEMU Virtual CPU version 2.5+                       0.00 GHz
 Dasharo (coreboot+UEFI) v0.2.0                      2048 MB RAM @ 200 MHz



   Select Language            <Standard English>

 > User Password Management
 > Device Manager
 > Dasharo System Features
 > One Time Boot
 > Boot Maintenance Manager

   Continue
   Reset





  LCtrl+LAlt+F12=Save screenshot
  ```

macpijan replied:
I suggest that we move it to a separate task, as getting the `Reset to defaults` to work on MSI is going to be tricky...

I suggest we stick with the original automation plan we've had in the configs, and focus on getting these to work, not adding more.

macpijan replied:
https://github.com/Dasharo/open-source-firmware-validation/issues/540

macpijan commented on lib/bios/menus.robot:
```diff
@@ -412,6 +413,48 @@ Reset To Defaults Tianocore
+
+Go To Dasharo Main Screen
+    [Documentation]    This keyword will press ESC until it will detect the main Dasharo main screen.
+    FOR    ${i}    IN RANGE    0    10
```
I tried and it needed 2 iterations to go back to the main menu with 1 nested screen, instead of one.

JanPrusinowski replied:
Maciej created a seprate task for this issue: https://github.com/Dasharo/open-source-firmware-validation/issues/540

macpijan commented on lib/bios/menus.robot:
```diff
@@ -412,6 +413,48 @@ Reset To Defaults Tianocore
+        Press Key N Times    3    ${ESC}
+    END
+
+Go To Dasharo Main Screen
```
You are aiming very high choosing to develop such a keyword.

It will be very tricky to get it right, depending on how it's been called.

In the first iteration this keyword relies on whatever was left in the telnet buffer by the previous keyword. The fact that you kewyord runs at all is simply because you chose to read until `save screenshot`, while in other menu handling keywords we read until `Esc=Exit` (`Save screenshot` has been added later at some point).

macpijan replied:
https://github.com/Dasharo/open-source-firmware-validation/issues/540

macpijan commented on platform-configs/include/msi-z690-common.robot:
```diff
@@ -86,6 +90,7 @@ ${MEMORY_PROFILE_SUPPORT}=                      ${TRUE}
 ${ESP_SCANNING_SUPPORT}=                        ${TRUE}
 ${DCU_UUID_SUPPORT}=                            ${TRUE}
 ${DCU_SERIAL_SUPPORT}=                          ${TRUE}
+${RESET_TO_DEFAULTS_SUPPORT}=                   ${TRUE}
```
That was disabled mostly because of the problems with serial being disabled after resetting. We can rather quickly test these manually after we finish automated tests.

macpijan replied:
see: https://github.com/Dasharo/open-source-firmware-validation/pull/530/files#r1806136564

JanPrusinowski replied:
This is a part of: https://github.com/Dasharo/open-source-firmware-validation/issues/540

macpijan commented on platform-configs/include/msi-z690-common.robot:
```diff
@@ -40,6 +40,9 @@ ${WIFI_CARD_UBUNTU}=                            ${TBD}
 ${USB_MODEL}=                                   Kingston
 ${USB_DEVICE}=                                  Multifunction Composite Gadget
 
+${INTEL_HYBRID_ARCH_SUPPORT}=                   ${FALSE}
```
If it relies on `rest to defaults`, we may give up on running these tests automatically right now as well: https://github.com/Dasharo/open-source-firmware-validation/pull/530/files#r1806136564

Or at least leave it until we have all of the "old" automation working in place.

macpijan replied:
https://github.com/Dasharo/open-source-firmware-validation/issues/540

macpijan commented on dasharo-compatibility/esp-scanning.robot:
```diff
@@ -27,6 +27,7 @@ Suite Setup         Run Keywords
 ...                     AND
 ...                     Prepare EFI Partition With System Files
 Suite Teardown      Run Keywords
+...                     Power On    AND
```
`Power On` is already in the `Clear Out EFI Partition` - seems redundant.

macpijan replied:
https://github.com/Dasharo/open-source-firmware-validation/blob/develop/lib/esp-scanning-lib.robot#L65

JanPrusinowski committed:
menus.robot: removed doubled state check in Reset to Defaults

Signed-off-by: Jan Prusinowski <jan.prusinowski@3mdeb.com>

JanPrusinowski commented on lib/bios/menus.robot:
```diff
@@ -407,11 +408,23 @@ Reset To Defaults Tianocore
+        ${dasharo_menu}=    Enter Dasharo System Features    ${menu}
+        ${serial_menu}=    Enter Dasharo Submenu    ${dasharo_menu}    Serial Port Configuration
+        ${serial_state}=    Get Option State    ${serial_menu}    Enable COM0 Serial
+        IF    ${serial_state} != ${TRUE}
```
I have removed the double check in: 49bc96d8b5835d5c5efc92929eeaa6e8248c23db

JanPrusinowski commented on lib/bios/menus.robot:
```diff
@@ -407,11 +408,23 @@ Reset To Defaults Tianocore
+        ${serial_menu}=    Enter Dasharo Submenu    ${dasharo_menu}    Serial Port Configuration
+        ${serial_state}=    Get Option State    ${serial_menu}    Enable COM0 Serial
+        IF    ${serial_state} != ${TRUE}
+            Set Option State    ${serial_menu}    Enable COM0 Serial    ${TRUE}
```
In each test I have come across which uses this KWD there is a 'Save and Reset' KWD used afterwards. As I understand pressing 'F9' wont save settings automatically. So there is a need to save setings anyway after using this kwd.

JanPrusinowski replied:
Enabling serial wont be merged with this branch. Instead it is a part of a seprate task: https://github.com/Dasharo/open-source-firmware-validation/issues/540

macpijan commented on dasharo-security/rebar.robot:
```diff
@@ -20,4 +20,4 @@ RBE001.001 Check if Resizeable BARs option is present
     Skip If    not ${DASHARO_PCIE_REBAR_SUPPORT}    RBE001.001 not supported
     ${out}=    Get UEFI Option    PCIeResizeableBarsEnabled
-    Should Contain    "${out}"    abled
+    IF    '${out}' not in ['True', 'False']    Fail    Option not found
```
Likely we need this to be ported to: https://github.com/Dasharo/open-source-firmware-validation/commits/regression-msi/

JanPrusinowski replied:
https://github.com/Dasharo/open-source-firmware-validation/commit/904dd1432f23058a795faff172a352c1ac3785a7

macpijan commented on dasharo-security/tcg-opal-disk-password.robot:
```diff
@@ -47,9 +47,9 @@ DMP001.001 TCG OPAL disk password set and check
     Save Changes And Reset
     @{password}=    Set Variable    1    2    3
-    Type In New Disk Password    @{password}
+    Type In New Disk Password    ${password}
```
@JanPrusinowski Likely we need this to be ported to: https://github.com/Dasharo/open-source-firmware-validation/commits/regression-msi/

JanPrusinowski replied:
https://github.com/Dasharo/open-source-firmware-validation/commit/edda9f798c950011a1acedf7cc46e905003570c3

macpijan commented on dasharo-security/verified-boot.robot:
```diff
@@ -159,7 +159,7 @@ VBO012.001 Self-signed binary is bootable without errors
     Clone Git Repository    https://github.com/Dasharo/dasharo-tools.git
     Execute Command In Terminal    rm -rf vboot_keys
-    ${out_genkey}=    Execute Command In Terminal    ./dasharo-tools/vboot/generate_keys vboot_keys    timeout=5m
+    ${out_genkey}=    Execute Command In Terminal    ./dasharo-tools/vboot/generate_keys vboot_keys    timeout=20m
```
@JanPrusinowski Likely we need this to be ported to: https://github.com/Dasharo/open-source-firmware-validation/commits/regression-msi/

krystian-hebel replied:
I was hoping that the log would show per-line timestamps, but SSH Library hides that information. Let's go with 20m then.

macpijan commented on keywords.robot:
```diff
@@ -212,7 +212,7 @@ Open Connection And Log In
     # FIXME: some stands do not have RTE connected, this should be better handled
     # by reworking variables.robot
-    IF    '${MANUFACTURER}' != 'QEMU'
+    IF    '${MANUFACTURER}' != 'QEMU' and '${CONFIG}' != 'no-rte'
```
50/50 - TBD?

macpijan commented on lib/options/uefi-setup-menu.robot:
```diff
@@ -59,7 +59,7 @@ Get UEFI Option
         ...    ${option_path[${i}]}
     END
 
-    ${state}=    Get Option State    ${menu}    ${option_path[${path_len}-1]}    ${VALUE}
```
@JanPrusinowski Likely we need this to be ported to: https://github.com/Dasharo/open-source-firmware-validation/commits/regression-msi/

BeataZdunczyk replied:
@JanPrusinowski link?

JanPrusinowski replied:
https://github.com/Dasharo/open-source-firmware-validation/commit/904dd1432f23058a795faff172a352c1ac3785a7

macpijan commented on platform-configs/msi-pro-z690-a-wifi-ddr4.robot:
```diff
@@ -3,17 +3,30 @@ Resource    include/msi-z690-common.robot
+${DMIDECODE_FIRMWARE_VERSION}=          Dasharo (coreboot+UEFI) v1.1.4-rc1
 ${DMIDECODE_PRODUCT_NAME}=              MS-7D25
-${DMIDECODE_RELEASE_DATE}=              11/27/2023
+${DMIDECODE_RELEASE_DATE}=              10/07/2024
```
@JanPrusinowski Likely we need this to be ported to: https://github.com/Dasharo/open-source-firmware-validation/commits/regression-msi/

JanPrusinowski replied:
https://github.com/Dasharo/open-source-firmware-validation/commit/c2aec8eb46036528534b53245960e8f184f8ddca

BeataZdunczyk commented:
Lots of threads to resolve. Could you please @macpijan , and @krystian-hebel recheck your threads so we can resolve them and merge this PR?

JanPrusinowski commented:
Everything needed from this PR was moved to: https://github.com/Dasharo/open-source-firmware-validation/pull/550
